prover/src/lib.rs

```diff
@@ -135,13 +136,23 @@ pub trait Prover {
     /// PRNG to be used for generating random field elements.
     type RandomCoin: RandomCoin<BaseField = Self::BaseField, Hasher = Self::HashFn>;
 
-    // Trace low-degree extension for building the LDEs of trace segments and their commitments.
+    /// Trace low-degree extension for building the LDEs of trace segments and their commitments.
     type TraceLde<E: FieldElement<BaseField = Self::BaseField>>: TraceLde<
         BaseField = Self::BaseField,
         ExtensionField = E,
         HashFn = Self::HashFn,
     >;
 
+    /// Constraints evaluator.
+    type ConstraintEvaluator<'a, E>: ConstraintEvaluator<
+        'a,
+        E,
+        Air = Self::Air,
+        TraceLde = Self::TraceLde<E>,
+    >
+    where
+        E: FieldElement<BaseField = Self::BaseField>;
```

Comment:
And this can become:
```Rust
type ConstraintEvaluator<'a, E>: ConstraintEvaluator<'a, E, Air = Self::Air>
where
    E: FieldElement<BaseField = Self::BaseField>;
```
Also, I'd change the comment to something like:
```
/// Constraints evaluator used to evaluate AIR constraints over the extended execution trace.
```